pymess/backend/dialer/daktela.py

```diff
@@ -0,0 +1,100 @@
+from chamber.utils.datastructures import ChoicesEnum
+from django.utils import timezone
+from django.utils.encoding import force_text
+from django.utils.translation import ugettext_lazy as _l
+
+from pymess.backend.dialer import DialerBackend
+from pymess.config import settings
+from pymess.models import DialerMessage
+from pymess.utils.logged_requests import generate_session
+
+
+class DaktelaDialerBackend(DialerBackend):
+    """
+    Dialer backend implementing Daktela service https://www.daktela.com/api/v6/models/campaignsrecords
+    """
+
+    STATE = ChoicesEnum(
+        ('NOT_ASSIGNED', _l('not assigned'), 'not assigned'),
+        ('READY', _l('ready'), 'ready'),
+        ('RESCHEDULED_BY_DIALER', _l('rescheduled by dialer'), 'rescheduled by dialer'),
+        ('CALL_IN_PROGRESS', _l('call in progress'), 'call in progress'),
+        ('HANGUP', _l('hangup'), 'hangup'),
+        ('DONE', _l('done'), 'done'),
+        ('RESCHEDULED', _l('rescheduled'), 'rescheduled'),
+    )
+
+    # TODO: clean-up mappings
+    STATES_MAPPING = {
+        STATE.NOT_ASSIGNED: DialerMessage.STATE.UNKNOWN,
+        STATE.READY: DialerMessage.STATE.READY,
+        STATE.RESCHEDULED_BY_DIALER: DialerMessage.STATE.RESCHEDULED,
+        STATE.CALL_IN_PROGRESS: DialerMessage.STATE.IN_PROGRESS,
+        STATE.HANGUP: DialerMessage.STATE.DONE,
+        STATE.DONE: DialerMessage.STATE.DONE,
+        STATE.RESCHEDULED: DialerMessage.STATE.RESCHEDULED,
+    }
+
+    @staticmethod
+    def _get_dialer_api_url(name=None):
+        base_url = '/'.join((
+            settings.DIALER_API_URL,
+            settings.DIALER_API_VERSION,
+            settings.DIALER_API_RESOURCE,
+        ))
+        name = '/' + name if name else ''
```

Comment:
co je name prosím, místo + raději používej format nebo to name dej do joinu a odfiltruj nulové hodnoty

Reply:
volanie dalsich daktela resources vyzera nejako takto `/api/v6/campaignsRecords/{NAME}.json`

`name` je parameter, ktory sa tam ma vlozit. kedze zakladna url vyzera `/api/v6/campaignsRecords.json`, meni sa to iba pomocou `/{NAME}`.

`name` teda bud posielam, alebo neposielam. preto je to takto...

Reply:
protože je to hůře čitelné, plete se to se sčítáním čísel. Navíc format rovnou obalí data str funkcí, takže nemusíš případně řešit přetypování

Reply:
mne sa to cita uplne v pohode, ale ok, prepisal som to na format